src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -295,13 +317,31 @@ object Directory {
 }
 
 // modifiedTime is an Epoch time in milliseconds. (ms since 1970-01-01T00:00:00.000 UTC).
-case class FileInfo(name: String, size: Long, modifiedTime: Long)
+// id, if set, is a unique identifier generated by Hyperspace, for internal use only.
```

Comment:
you can remove "if set" since `id` is always being set (not `Option`)? Btw, can we use `Option` instead or no? Or can we always set it to a valid id or no?

Reply:
We assign id to a file once we index it. This avoids having gaps in the id (For example adding a file as an appendedFile and removing it before refresh). I think using UNKNOWN_FILE_ID is clear enough. I am not sure how an Option would help here. I prefer to keep it like this, due to simplicity and clarity. Please let me know if you still suggest a change here.

Reply:
How about this question "Or can we always just set it to a valid id or no?" 

Why not just assign id to each file instead of exposing -1?

Reply:
> I am not sure how an Option would help here.

Avoids having a magic number.

> This avoids having gaps in the id

Is this a requirement? No, right?

Reply:
Having no gap is not a requirement and a gap will happen if a user deletes some files.
If we force each/every FileInfo to have an id, then: 
1. We need to add code to generate id even for index files. However, id in the context of an index file has no use. 
2. We dont assign an id to an appended file until we index it. This semantically makes sense as usage of id is for capturing lineage and naturally it should be generated/assigned once records from the file start showing up in the index.

The notion of FileInfo has a larger scope in the code-base than just capturing source data files and the id is only meaningful and relevant in the context of a source data file. Using a default value for a file which is not id-ed yet is natural to me. 

With the Option suggestion, are you suggestion making id filed of type Option[Long] in the definition of `FileInfo` case class?

Reply:
> * We need to add code to generate id even for index files. However, id in the context of an index file has no use.

What will be the overhead?

> * We dont assign an id to an appended file until we index it. This semantically makes sense as usage of id is for capturing lineage and naturally it should be generated/assigned once records from the file start showing up in the index.

Isn't it more clearer if all files have the ids? (e.g., no need to explain about why some files have -1). To me, the usage of id is not to capture the lineage, but an optimization to represent a "hash" of a full path.

Reply:
If we assign id to each `FileInfo`, does the overall code get better or worse?

> With the Option suggestion, are you suggestion making id filed of type Option[Long] in the definition of `FileInfo` case class?

Yea, and it will show `null` instead of `-1`.

Reply:
> 2\. This semantically makes sense as usage of id is for capturing lineage

So, does this mean we need to set it to `-1` if `spark.hyperspace.index.lineage.enabled` is false?

Reply:
`-1` is some number, picked to show Unknown/Unassigned id. It can be null, if we go with the `Option` approach. This is just an indicator that Hyperspace did not generated/assigned an id to this file (yet). In case of a source data file, this happens for an appended file and the file gets an id once it is indexed. In case of an index file, it never gets an id because it does not need it.
Just to make sure I understand your point here: Is your concern about using `-1` vs. `null`, to show an unknown/unassigned file id ?

Reply:
My point is if `id` is optional, let's use `Option`.